Microsoft.HealthVault.Fhir/Transformers/PersonItemToFhir.cs

```diff
@@ -0,0 +1,170 @@
+            {
+                var fhirAddress = new Hl7.Fhir.Model.Address
+                {
+                    Text = person.Organization,
+                    Use = Hl7.Fhir.Model.Address.AddressUse.Work
+                };
+                practitioner.Address.Add(fhirAddress);
+                //practitioner.AddExtension(HealthVaultExtensions.PersonOrganisation, new FhirString(person.Organization));
+            }
+
+            if (!string.IsNullOrEmpty(person.ProfessionalTraining))
+            {
+                var qualificationComponent = new Practitioner.QualificationComponent
+                {
+                    Code = new CodeableConcept
+                    {
+                        Text = person.ProfessionalTraining
+                    }
+                };
+                practitioner.Qualification = new System.Collections.Generic.List<Practitioner.QualificationComponent>
+                {
+                    qualificationComponent
+                };
+            }
+
+            if (!string.IsNullOrEmpty(person.PersonId))
+            {
+                practitioner.Identifier = new System.Collections.Generic.List<Identifier>
+                {
+                   new Identifier
+                   {
+                       Value = person.PersonId
+                   }
+                };
+            }
+
+            if (person.ContactInformation != null)
+            {
+                practitioner.Address.AddRange(
+                    person.ContactInformation.Address.Select(address => ToFhirInternal(address)));
+                practitioner.Telecom.AddRange(
+                    person.ContactInformation.Phone.Select(phone => ToFhirInternal(phone)));
+                practitioner.Telecom.AddRange(
+                    person.ContactInformation.Email.Select(email => ToFhirInternal(email)));
+            }
+
+            return practitioner;
+        }
+
+        private static HumanName ToFhirInternal(Name hvName)
+        {
+            var fhirName = new HumanName
+            {
+                Text = hvName.Full,
+                Family = hvName.Last
+            };
+
+            if (hvName.Title != null)
+            {
+                fhirName.AddExtension(HealthVaultExtensions.PatientTitle, hvName.Title.ToFhir());
+            }
+
+            AddGivenName(fhirName, hvName.First);
+            AddGivenName(fhirName, hvName.Middle);
+
+            if (hvName.Suffix != null)
+            {
+                fhirName.AddExtension(HealthVaultExtensions.PatientSuffix, hvName.Suffix.ToFhir());
+            }
+
+            return fhirName;
+        }
+
+        private static void AddGivenName(HumanName fhirName, string first)
+        {
+            if (!string.IsNullOrEmpty(first))
+            {
+                fhirName.GivenElement.Add(new FhirString(first));
+            }
+        }
+
+        private static Hl7.Fhir.Model.Address ToFhirInternal(ItemTypes.Address hvAddress)
```

Comment:
Can you abstract this out to where it can also be used in "Transformers/ContactToFhir.cs"? There's very similar code there.

Reply:
Fixed; Added a new ToFhir extension for ItemTypes.Address and the same is used in both PersonItem and Contact transformer classes